src/main/scala/julienrf/json/derived/FieldAdapter.scala

```diff
@@ -0,0 +1,25 @@
+package julienrf.json.derived
+
+trait Adapter extends (String => String)
+
+object Adapter {
+
+  val snakeCase = Adapter {
+    (s: String) => {
+      s.foldLeft(new StringBuilder) {
+        case (s, c) if Character.isUpperCase(c) && s.nonEmpty => s append "_" append (Character toLowerCase c)
+        case (s, c) => s append c
+      }.toString
```

Comment:
I find it weird to mix FP and imperative style. Why not just use a `foreach` instead of a `foldLeft`?